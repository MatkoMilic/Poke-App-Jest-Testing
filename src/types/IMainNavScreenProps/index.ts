export type {
  IMainNavScreenProps,
  MainNavigationType,
} from './IMainNavScreenProps';
