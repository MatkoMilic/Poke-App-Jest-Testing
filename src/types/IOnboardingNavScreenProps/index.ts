export type {
  IOnboardingNavScreenProps,
  OnboardingNavigationType,
} from './IOnboardingNavScreenProps';
