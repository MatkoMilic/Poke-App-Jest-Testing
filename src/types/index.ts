export type {
  IMainNavScreenProps,
  MainNavigationType,
} from './IMainNavScreenProps';
export type {
  IOnboardingNavScreenProps,
  OnboardingNavigationType,
} from './IOnboardingNavScreenProps';
export type {
  MainNavigatorKeys,
  NavigatorNameKeys,
  OnboardingNavigatorParamList,
  RootNavigatorKeys,
  RootNavigatorParamList,
  OnboardingNavigatorKeys,
  MainNavigatorParamList,
  RootNavigatorType,
} from './navigatorTypes';
export {
  NavigatorNames,
  OnboardingNavigatorScreens,
  MainNavigatorScreens,
} from './navigatorTypes';
export type {IUserValues} from './IUserValues/IUserValues';
export {initialUserLoginValues} from './IUserValues/IUserValues';
