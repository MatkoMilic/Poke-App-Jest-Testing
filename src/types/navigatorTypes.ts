import {NavigatorScreenParams} from '@react-navigation/native';

export enum OnboardingNavigatorScreens {
  LOADING_SCREEN = 'LoadingScreen',
  LOGIN_SCREEN = 'LoginScreen',
}
export type OnboardingNavigatorParamList = {
  LOADING_SCREEN: undefined;
  LOGIN_SCREEN: undefined;
};
export type OnboardingNavigatorKeys = keyof OnboardingNavigatorScreens;
export enum MainNavigatorScreens {
  PROFILE_SCREEN = 'ProfileScreen',
  POKELIST_SCREEN = 'PokeListScreen',
  SETTINGS_SCREEN = 'SettingsScreen',
}
export type MainNavigatorParamList = {
  PROFILE_SCREEN: undefined;
  POKELIST_SCREEN: undefined;
  SETTINGS_SCREEN: undefined;
};
export type MainNavigatorKeys = keyof MainNavigatorScreens;
export enum NavigatorNames {
  MainNavigator = 'MainNavigator',
  OnboardingNavigator = 'OnboardingNavigator',
}
export type NavigatorNameKeys = keyof NavigatorNames;
export type RootNavigatorType = {
  MainNavigator: NavigatorScreenParams<MainNavigatorParamList>;
  OnboardingNavigator: NavigatorScreenParams<OnboardingNavigatorParamList>;
};
export type RootNavigatorParamList = {
  MainNavigator: undefined;
  OnboardingNavigator: undefined;
};
export type RootNavigatorKeys = keyof RootNavigatorType;
