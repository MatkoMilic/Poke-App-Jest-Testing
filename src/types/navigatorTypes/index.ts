export type {
  MainNavigatorKeys,
  OnboardingNavigatorKeys,
  MainNavigatorParamList,
  RootNavigatorParamList,
  RootNavigatorType,
  RootNavigatorKeys,
  NavigatorNameKeys,
  OnboardingNavigatorParamList,
} from './navigatorTypes';
export {
  NavigatorNames,
  OnboardingNavigatorScreens,
  MainNavigatorScreens,
} from './navigatorTypes';
