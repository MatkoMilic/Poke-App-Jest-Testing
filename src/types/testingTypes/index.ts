export type {ReactWrapperType, RenderApiType} from './testingTypes';
