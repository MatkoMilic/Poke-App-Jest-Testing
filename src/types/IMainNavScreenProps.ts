import {CompositeNavigationProp} from '@react-navigation/native';
import {NativeStackNavigationProp} from '@react-navigation/native-stack';
import {RootNavigatorType, MainNavigatorParamList} from './navigatorTypes';

export type MainNavigationType = CompositeNavigationProp<
  NativeStackNavigationProp<MainNavigatorParamList>,
  NativeStackNavigationProp<RootNavigatorType>
>;
export interface IMainNavScreenProps {
  navigation: MainNavigationType;
}
