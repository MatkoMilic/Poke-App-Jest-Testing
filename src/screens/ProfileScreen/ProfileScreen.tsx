import React from 'react';
import {Text} from 'react-native';
import {Header, ScreenContainer} from '../../components';
import {IMainNavScreenProps, MainNavigatorScreens} from '../../types';
import {PokeListScreen} from '../PokeListScreen';

interface ProfileScreenProps extends IMainNavScreenProps {}

const ProfileScreen: React.FC<ProfileScreenProps> = ({navigation}) => {
  const goToSettings = () => {
    navigation.navigate(MainNavigatorScreens.SETTINGS_SCREEN);
  };
  const goToPokeList = () => {
    navigation.navigate(MainNavigatorScreens.POKELIST_SCREEN);
  };
  return (
    <ScreenContainer>
      <Header
        headerTitle="Poke Profile"
        leftIcon="account-cog"
        rightIcon="clipboard-list"
        headerSubtitle="2front"
        leftOnPress={goToPokeList}
        rightOnPress={goToSettings}
      />
      <Text>Welcome to profile screen</Text>
    </ScreenContainer>
  );
};

export default ProfileScreen;
