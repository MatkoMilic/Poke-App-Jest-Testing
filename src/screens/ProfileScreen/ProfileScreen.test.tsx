import React from 'react';
import renderer from 'react-test-renderer';
import {MainNavigationType} from '../../types';
import ProfileScreen from './ProfileScreen';

let navigation: MainNavigationType;

test('renders correctly', () => {
  const tree = renderer
    .create(<ProfileScreen navigation={navigation} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
