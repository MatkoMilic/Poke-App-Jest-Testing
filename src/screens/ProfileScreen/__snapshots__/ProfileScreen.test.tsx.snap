// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaView
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <Text>
    Welcome to profile screen
  </Text>
</RNCSafeAreaView>
`;
